ESSArch_Core/fixity/validation/tests/test_xml_schema_validator.py

```diff
@@ -0,0 +1,521 @@
+import uuid
+import os
+import tempfile
+import shutil
+
+from unittest import mock
+from django.test import TestCase
+from lxml.etree import DocumentInvalid
+
+from ESSArch_Core.fixity.validation.backends.xml import (
+    XMLSchemaValidator,
+    XMLSchematronValidator,
+    XMLISOSchematronValidator,
+)
+
+
+def create_mocked_error_log(line_number, error_msg):
+    class MockedErrorLog:
+        line = line_number
+        message = error_msg
+
+    return MockedErrorLog()
+
+
+class XMLSchemaValidatorTests(TestCase):
+
+    @mock.patch("ESSArch_Core.fixity.validation.backends.xml.Validation.objects.bulk_create")
+    @mock.patch("ESSArch_Core.fixity.validation.backends.xml.validate_against_schema")
+    def test_when_documentInvalid_raised_create_validation_objects(self, validate_against_schema, validation_obj):
+        exception_obj = DocumentInvalid("error msg")
+        exception_obj.error_log = [
+            create_mocked_error_log(123, "error msg 1"),
+            create_mocked_error_log(456, "error msg 2"),
+        ]
+        validate_against_schema.side_effect = mock.Mock(side_effect=exception_obj)
+
+        ip_id = uuid.uuid4()
+        task_id = uuid.uuid4()
+
+        validator = XMLSchemaValidator(
+            context="dummy_schema.xsd",
+            options={'rootdir': "dummy_rootdir"},
+            ip=ip_id,
+            task=task_id
+        )
+        with self.assertRaises(DocumentInvalid):
+            validator.validate("some_xml_file_path")
+
+        validation_obj.assert_called_once_with([mock.ANY, mock.ANY], 100)
```

Comment:
Instead of checking how `bulk_create` and `create` is called, check which items have been created. This should be the case for all tests in this module

Reply:
will do!